code/modules/mob/living/carbon/human/examine.dm

```diff
@@ -364,13 +364,22 @@
 /proc/hasHUD(mob/M as mob, hudtype)
 	if(istype(M, /mob/living/carbon/human))
 		var/mob/living/carbon/human/H = M
-		switch(hudtype)
-			if("security")
-				return istype(H.glasses, /obj/item/clothing/glasses/hud/security) || istype(H.glasses, /obj/item/clothing/glasses/sunglasses/sechud)
-			if("medical")
-				return istype(H.glasses, /obj/item/clothing/glasses/hud/health)
-			else
-				return 0
+		var/obj/item/clothing/glasses/G = H.glasses
+		var/eye_hud = 0
+		var/hud = 0
+		// Checks for eye sensor HUD
+		var/obj/item/organ/internal/augment/eye_sensors/E = locate() in H.internal_organs
+		if(E)
+			eye_hud = E.check_hud(hudtype)
+		if(G)
+			switch(hudtype)
+				if("security")
+					hud = G.is_sec_hud()
+				if("medical")
+					hud = G.is_med_hud()
+				else
+					return 0
```

Comment:
```suggestion
```

Wearing hudless glasses will block hud-eye hud-vision if these lines are kept.

Reply:
That doesn't do that, it means that if the proc is called (which currently is either to check if security or medical HUDs are active) and the requested hud type wasn't security or medical, it stops, but its a redundant line anyway.

Reply:
Ah, right. Thanks 👍